dev-libs/keystone/keystone-0.9.1.ebuild

```diff
@@ -0,0 +1,81 @@
+# Copyright 1999-2017 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+DISTUTILS_OPTIONAL=1
+CMAKE_MIN_VERSION=3.8.7
+PYTHON_COMPAT=( python{2_7,3_4,3_5,3_6} )
+
+inherit cmake-utils distutils-r1
+
+DESCRIPTION="assembly/assembler framework + bindings"
+HOMEPAGE="http://www.keystone-engine.org/"
+
+if [[ ${PV} == 9999* ]]; then
+	EGIT_REPO_URI="https://github.com/keystone-engine/keystone.git"
+	inherit git-r3
+	KEYWORDS=""
+else
+	SRC_URI="https://github.com/keystone-engine/keystone/archive/${PV/_rc/-rc}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm ~x86"
+fi
+
+LICENSE="GPL-2"
+SLOT="0"
+
+# Keep in sync with llvm/CMakeLists.txt, subset of sys-devel/llvm
+ALL_LLVM_TARGETS=( AArch64 ARM Hexagon Mips PowerPC Sparc SystemZ X86 )
+ALL_LLVM_TARGETS=( "${ALL_LLVM_TARGETS[@]/#/llvm_targets_}" )
+LLVM_TARGET_USEDEPS=${ALL_LLVM_TARGETS[@]/%/?}
+
+IUSE="python ${ALL_LLVM_TARGETS[*]}"
+RDEPEND="sys-devel/llvm:=[${LLVM_TARGET_USEDEPS// /,}]
+	${PYTHON_DEPS}
+"
+DEPEND="${RDEPEND}"
+REQUIRED_USE="
+	${PYTHON_REQUIRED_USE}
+	|| ( ${ALL_LLVM_TARGETS[*]} )
+"
+
+S=${WORKDIR}/${P/_rc/-rc}
+
+CMAKE_BUILD_TYPE=RelWithDebInfo
+
+wrap_python() {
+	if use python; then
+		pushd bindings/python >/dev/null || die
+		distutils-r1_${1} "$@"
```

Comment:
You can use `${EBUILD_PHASE_FUNC}` to avoid having to pass it all over the place.